swingset/src/main/java/com/nqadmin/swingset/formatting/SSDateFormatterFactory.java

```diff
@@ -49,71 +50,118 @@
 // SwingSet - Open Toolkit For Making Swing Controls Database-Aware
 
 /**
- * SSDateFormatterFactory extends DefaultFormatterFactory for Date fields.
+ * Static methods to create a DefaultFormatterFactory for Date fields.
  */
-public class SSDateFormatterFactory extends javax.swing.text.DefaultFormatterFactory {
+public class SSDateFormatterFactory {
+
+	private SSDateFormatterFactory(){}
 
 	/**
 	 * Constant for dd/MM/yyyy date format
+	 * @deprecated 
 	 */
+	@Deprecated
 	public static final int DDMMYYYY = 1;
 
 	/**
 	 * Log4j Logger for component
 	 */
-	private static Logger logger = LogManager.getLogger();
+	private static final Logger logger = LogManager.getLogger();
 
 	/**
 	 * Constant for MM/dd/yyyy date format
+	 * @deprecated 
 	 */
+	@Deprecated
 	public static final int MMDDYYYY = 0;
 
 	/**
-	 * Unique serial ID
+	 * Constant for yyyy-MM-dd date format
+	 * @deprecated 
 	 */
-	private static final long serialVersionUID = -8205600502325364394L;
+	@Deprecated
+	public static final int YYYYMMDD = 2;
 
 	/**
-	 * Constant for yyyy-MM-dd date format
+	 * Create formatter factory with default pattern.
+	 * @return
 	 */
-	public static final int YYYYMMDD = 2;
+	// Don't like the way default is buried in get(Format) case label.
```

Comment:
Right, left over comment before adding Format.DATE to signify default.

The single enum approach would allow something like `SSFormattedTextField.get(Format.DATE)` to return a text field with default formatter for the given type.  Another reason is to make it easier to specify the default per app, rather than spreading it around in the different places FormatterFactory.

I think you mentioned somewhere that there could be an enum that just has the types like DATE/MONEY/... and then separate enums for each sub-type. That would work as well, the default sub-type would be specified as part of the type enum; that does feel cleaner and gives more compile time checking. It also gives a single spot for specifying the default sub-type for a given type.

After all the other issues are addressed, I'll put together a change just for the enums. Please speak up with any naming and/or other conventions in this area.

Reply:
In com.nqadmin.swingset.enums we use mixed case class enum name and all caps items, which is what you were doing with FORMAT enum. I guess we could consider a .formatting.enus package, but that might be overkill. Should we try to use the delimiter in the item names? YYYYMMDD_DELIM_HYPEN? Probably not. Other formatters might have multiple delimiters/separators (e.g., currency might have "." for cents and "," for thousands). Just trying to list some things that might cause enum ambiguity or confusion if we're going to handle order and delimiters, etc. in a single enum item.

Reply:
One possibility (blue sky) would be to only have FieldType enum. Then something like

- `SSFormattedTextField DateField.get()` returns something with reasonable defaults
- `SSFormattedTextField DateField.builder().mask("xxx").delimiter("").validator(doVal).build()` if you don't want the defaults

Then you don't need any enums. But this might be to open ended especially considering that it has to work with the verifier; I guess the stuff like DATE_YYYYDDMM sets up a variety of things so it all works together, and is a convenience. But in general, each subType may have it's own set of options and that complicates things again.

It might be a good idea to consider/try a few more Type implementations. I'm getting confused as to what's needed.